src/environment/environment.js

```diff
@@ -0,0 +1,127 @@
+/** @flow */
+import os from 'os';
+import { v4 } from 'uuid';
+import fs from 'fs-extra';
+import path from 'path';
+import npmInstall from '../utils/npm';
+import { loadScope } from '../scope';
+import { flattenDependencies } from '../scope/flatten-dependencies';
+import { BitId } from '../bit-id';
+import { Component } from '../consumer/component/consumer-component'
+import { BITS_DIRNAME } from '../constants';
+import { mergeAll } from 'ramda';
+
+const root = path.join(os.tmpdir(), 'bit');
+const currentPath = process.cwd();
+
+export default class Environment {
+  path: string;
+  component: Component;
+  componentsDependencies: ConsumerComponent[] = [];
+
+  constructor() {
+    this.path = path.join(root, v4());
+    console.log(this.path);
+  }
+
+  init(): Promise<*> {
+    return new Promise((resolve, reject) => {
+      fs.ensureDir(this.path, (err) => {
+        if (err) return reject(err);
+        return resolve();
+      });
+    });
+  }
+
+  importById(rawId: string): Promise<Object> {
+    // loadScope works for local scope and remote scope
+    return loadScope(currentPath).then((scope) => {
+      const bitId = BitId.parse(rawId, scope.name);
+      return scope.get(bitId)
+        .then((component) => {
+          component.component.write(this.path, true, true);
+          this.component = component.component;
+          return component;
+        });
+    });
+  }
+
+  writeToEnvironmentDir(componentDependencies: versionDependencies[]): Promise<Component[]> {
+    const componentsDir = path.join(this.path, BITS_DIRNAME);
```

Comment:
It would be better to reuse the consumer code for this task, otherwise, it is duplicated/different when it should imitate it.

It'll be probably better to just create a consumer, and work with the consumer methods.
I would have put the created consumer reference in the state of the class. (after init method.)